pycir/cir.py

```diff
@@ -0,0 +1,97 @@
+# -*- coding: utf-8 -*-
+"""
+Centered isotonic regression, generic but more tailored to dose-response and dose-finding data
+
+Assaf Oron, recoded from R 'cir' package
+"""
+
+import numpy as np
+import pandas as pd
+
+
+def cirPAVA(y, x=None, wt=None, outx=None, interiorStrict=True, strict=False, ybounds=None, full=False, dec=False):
+    """
+    Perform regression.
+
+    Args:
+        x,y,wt: vectors of equal length with the doses, mean y values (usually response rates in [0,1]), and weights (often sample size)
+        outx: ...
+    """
+    if ybounds is None:
+        ybounds = np.asarray([0.0,1.0])
+
+    y = np.asarray(y)
+    if y.ndim != 1:
+        raise Exception(f'You supplied an array with {y.ndim} dimensions, but must be a vector (i.e. 1 dimension)')
```

Comment:
Below you assume that `y` is a 1D array, so good to add a check and ensure it actually is.